app/services/audit/event/util.rb

```diff
@@ -0,0 +1,18 @@
+module Audit
+  module Event
+    class Util
+      class << self
+        def deserialize(string)
+          ActiveSupport::JSON.decode(string).deep_symbolize_keys.tap do |h|
+            h[:time] = Time.iso8601(h[:time])
```

Comment:
This should be `Time.zone.iso8601` or `Time.zone.parse` if there's any chance the time string is not ISO

Reply:
For my custom object serialization, I borrow this implementation from Rails. Starting from Rails 6.0 there is already such implementation.
https://github.com/rails/rails/commit/d9a5c7011f62dd771a2fa430090e068b1f9785f3

Today I notice, they have implementation also for `ActiveSupport::TimeWithZone`, but I am using `Time` as class in the serialization.

Also as side note, It would be helpful if I introduce some kind of deprecation warning, for this `Util` class. Only if this project comes to Rails 6.

Reply:
I think what they're doing make sense, the objective of a deserializer is not to change the data but in this case we should convert dates and times in a single timezone, so that data is stored with a single TZ.

For example:

```ruby
[3] pry(main)> Time.iso8601 "2014-12-04T11:02:37-05:00"
=> 2014-12-04 11:02:37 -0500
[4] pry(main)> Time.zone.iso8601 "2014-12-04T11:02:37-05:00"
=> Thu, 04 Dec 2014 16:02:37 UTC +00:00
```

Both are correct, it just depends on what we want to have at runtime

Reply:
Ok. I'll change this. Lets keep the consistency of Time entries in db.